internal/pubsub/pubsub.go

```diff
@@ -0,0 +1,47 @@
+package pubsub
+
+import (
+	"context"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/bitrise-io/bitrise-webhooks/service/hook/common"
+	"github.com/pkg/errors"
+	"google.golang.org/api/option"
+)
+
+// Client ...
+type Client struct {
+	c             *pubsub.Client
+	pubsubTopicID string
+}
+
+// NewClient ...
+func NewClient(projectID, serviceAccountJSON, pubsubTopicID string) (*Client, error) {
+	client, err := pubsub.NewClient(context.Background(), projectID, option.WithCredentialsJSON([]byte(serviceAccountJSON)))
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	return &Client{c: client, pubsubTopicID: pubsubTopicID}, nil
+}
+
+// PublishMetrics ...
+func (c *Client) PublishMetrics(metrics common.Metrics) (err error) {
+	if c == nil {
+		return nil
```

Comment:
Good question, I don't think returning an error would be the best, because this is a coding issue. You shouldn't call anything on a nil Client (but we still wanted to make sure this won't crash the app, I also have seen a similar pattern in the go-github repo, for example, in the [PushEvent.GetRepo](https://github.com/google/go-github/blob/master/github/github-accessors.go#L16906)).

Instead of returning an error, I would probably leave it as it is, or simply remove the nil check, what do you think?

Reply:
I would simply remove the nil check, BUT because it has a pointer receiver a nil value can still be passed, so lets leave it as it is.